File: client/src/app/common/purchase.ts
import { Addres } from './addres';
import { Customer } from './customer';
import { Order } from './order';
import { OrderItem } from './order-item';

export class Purchase {
  public customer: Customer | undefined;
  public shippingAddress: Addres | undefined;
  public billingAddress: Addres | undefined;
  public order: Order | undefined;
  public orderItems: OrderItem[] = [];
  constructor() {}
}

File: ecommerce/src/main/java/be/yt/ecommerce/dto/Purchase.java
package be.yt.ecommerce.dto;

import be.yt.ecommerce.entity.Address;
import be.yt.ecommerce.entity.Customer;
import be.yt.ecommerce.entity.Order;
import be.yt.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}
